staging/src/k8s.io/component-base/metrics/testutil/metrics.go

```diff
@@ -265,17 +262,26 @@ func GetHistogramVecFromGatherer(gatherer metrics.Gatherer, metricName string, l
 		}
 	}
 
-	if metricFamily == nil {
```

Comment:
it could still be `nil` (if no metric matches the name)

Reply:
yes, as you said the `len` check has covered this, see your comments here :).

https://github.com/kubernetes/kubernetes/pull/96258#discussion_r679949176

Reply:
you can remove `metricFamily.GetMetric() == nil`, but not this check